### ISSUE TITLE
Run reporter in browser

### ISSUE THREAD

**Issue by aleksandr-oleynikov** [travisjeffery/mocha-teamcity-reporter#41]
Hello!

Is there any legal way to run `mocha-teamcity-reporter` in browser?
Simple adding
```html
<script src="node_modules/mocha-teamcity-reporter/lib/teamcity.js"></script>
```
obviously leads to errors.

We test video players' iframes. In this way we run our tests in browser.
And we've set transferring log messages from browser console to node script, so we only need to get teamcity log messages in browser console.

**Comment by jamie-sherriff** [comment]
This should work but unsure of the correct setup method, what browser are you using? as I would say I broke some browser compatibility with the version 2.x release. (Feel free to confirm against 1.x) Can you post the errors you are getting?

**Comment by jamie-sherriff** [comment]
@aleksandr-oleynikov  yea... i can see how this doesn't work anymore using node related functions like process.env, sorry about that.

As an easy fix id say stick to using version 1 and ill come up with a better solution in the future.

**Comment by jamie-sherriff** [comment]
Actually if you can give my quick attempt at solving this a try
https://github.com/travisjeffery/mocha-teamcity-reporter/tree/run-in-browser

**Comment by aleksandr-oleynikov** [comment]
Hello @jamie-sherriff! Sorry for late answer.

I've collected some info which I hope can help with fixing this problem.

----------------------------------------------------------------------

At first I've tried to use current last released **version 2.3.0** in browser. It doesn't work.
<details>
<summary>Adding `<script src="node_modules/mocha-teamcity-reporter/lib/teamcity.js"></script>` to a page</summary>

1. Simple adding leads to an error that module `util` cannot be found
<details>
  <summary>Screenshot</summary>

![2 3 0-script-1 util](https://user-images.githubusercontent.com/22796829/38987936-27e35812-43da-11e8-8775-f77d42901cac.png)
</details>

2. I downloaded [util](https://www.npmjs.com/package/util) module and put it in global scope (`window.util = util;`). So I could delete the line `const util = require('util');` in `teamcity.js`.
When I did it I got another error about the absence of `process`
<details>
  <summary>Screenshot</summary>

![2 3 0-script-2 process](https://user-images.githubusercontent.com/22796829/38988284-53a2bb90-43db-11e8-9c34-0d2fe3941ced.png)
</details>
</details>

<details>
  <summary>Bundling with webpack</summary>

  1. When I tried bundling my build with imported `mocha-teamcity-reporter` I got an error about the absence of `./base` module which is really absent
<details>
  <summary>webpack log</summary>

```
> video-players-api@0.1.0 build /Users/al88/AL/Dev/video-players-api
> webpack

clean-webpack-plugin: /Users/al88/AL/Dev/video-players-api/dist has been removed.
Hash: acd13c40cc43686307cd
Version: webpack 4.1.0
Time: 2913ms
Built at: 2018-4-19 13:18:17
        Asset      Size  Chunks                    Chunk Names
    bundle.js   1.1 MiB    main  [emitted]  [big]  main
bundle.js.map  1.04 MiB    main  [emitted]         main
Entrypoint main [big] = bundle.js bundle.js.map
[./node_modules/webpack/buildin/global.js] (webpack)/buildin/global.js 509 bytes {main} [built]
[./test/cases.js] 684 bytes {main} [built]
[./test/config/index.js] 6.54 KiB {main} [built]
[./test/config/rutube.js] 828 bytes {main} [built]
[./test/helpers/listener.js] 6.93 KiB {main} [built]
[./test/helpers/loadScript.js] 398 bytes {main} [built]
[./test/helpers/logger.js] 2.97 KiB {main} [built]
[./test/helpers/sender.js] 2.98 KiB {main} [built]
[./test/helpers/shouldSkip.js] 563 bytes {main} [built]
   [0] multi babel-polyfill ./test/setup ./test/run 52 bytes {main} [built]
[./test/helpers/waitUntil.js] 1.5 KiB {main} [built]
[./test/hooks.js] 1.82 KiB {main} [built]
[./test/run.js] 809 bytes {main} [built]
[./test/setup.js] 3.39 KiB {main} [built]
[./test/specs/index.js] 315 bytes {main} [built]
    + 426 hidden modules

WARNING in ./node_modules/mocha/lib/mocha.js
153:20-37 Critical dependency: the request of a dependency is an expression
 @ ./node_modules/mocha/lib/mocha.js
 @ ./node_modules/mocha/browser-entry.js
 @ ./node_modules/mocha-teamcity-reporter/lib/teamcity.js
 @ ./test/setup.js
 @ multi babel-polyfill ./test/setup ./test/run

WARNING in ./node_modules/mocha/lib/mocha.js
158:24-70 Critical dependency: the request of a dependency is an expression
 @ ./node_modules/mocha/lib/mocha.js
 @ ./node_modules/mocha/browser-entry.js
 @ ./node_modules/mocha-teamcity-reporter/lib/teamcity.js
 @ ./test/setup.js
 @ multi babel-polyfill ./test/setup ./test/run

WARNING in ./node_modules/mocha/lib/mocha.js
193:17-30 Critical dependency: the request of a dependency is an expression
 @ ./node_modules/mocha/lib/mocha.js
 @ ./node_modules/mocha/browser-entry.js
 @ ./node_modules/mocha-teamcity-reporter/lib/teamcity.js
 @ ./test/setup.js
 @ multi babel-polyfill ./test/setup ./test/run

WARNING in ./node_modules/mocha/lib/mocha.js
231:26-39 Critical dependency: the request of a dependency is an expression
 @ ./node_modules/mocha/lib/mocha.js
 @ ./node_modules/mocha/browser-entry.js
 @ ./node_modules/mocha-teamcity-reporter/lib/teamcity.js
 @ ./test/setup.js
 @ multi babel-polyfill ./test/setup ./test/run

ERROR in ./node_modules/mocha-teamcity-reporter/lib/teamcity.js
Module not found: Error: Can't resolve './base' in '/Users/al88/AL/Dev/video-players-api/node_modules/mocha-teamcity-reporter/lib'
 @ ./node_modules/mocha-teamcity-reporter/lib/teamcity.js 31:8-25
 @ ./test/setup.js
 @ multi babel-polyfill ./test/setup ./test/run

ERROR in ./node_modules/mkdirp/index.js
Module not found: Error: Can't resolve 'fs' in '/Users/al88/AL/Dev/video-players-api/node_modules/mkdirp'
 @ ./node_modules/mkdirp/index.js 2:9-22
 @ ./node_modules/mocha/lib/reporters/xunit.js
 @ ./node_modules/mocha/lib/reporters/index.js
 @ ./node_modules/mocha/lib/mocha.js
 @ ./node_modules/mocha/browser-entry.js
 @ ./node_modules/mocha-teamcity-reporter/lib/teamcity.js
 @ ./test/setup.js
 @ multi babel-polyfill ./test/setup ./test/run
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! video-players-api@0.1.0 build: `webpack`
npm ERR! Exit status 2
npm ERR!
npm ERR! Failed at the video-players-api@0.1.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/al88/.npm/_logs/2018-04-19T10_18_17_572Z-debug.log
```
</details>

2. After that I deleted all unnecessary `require`s and left only these lines at the beginning of `teamcity.js`:
```js
  // running in browser (possibly phantomjs) but without require
  Base = window.Mocha.reporters.Base
  log = console.log
```
This leads to successful bundle creating but an error about `process.pid` on test page.
<details>
  <summary>Screenshot</summary>

  
![2 3 0-webpack-2 process pid](https://user-images.githubusercontent.com/22796829/38988763-19f0f964-43dd-11e8-980c-3ea28c272707.png)
</details>
</details>

----------------------------------------------------------------------

Unfortunately, using **branch [run-in-browser](https://github.com/travisjeffery/mocha-teamcity-reporter/tree/run-in-browser)** didn't fix this problem. The errors are exactly the same as for version 2.3.0.

----------------------------------------------------------------------

**Version 1.1.0** of `mocha-teamcity-reporter` worked when I had added its `teamcity.js` script to our test page: 
```
<script src="node_modules/mocha-teamcity-reporter/lib/teamcity.js"></script>
```

FYI Bundling this version with webpack doesn't work. I get an error about the absence of `./base` module similar to corresponding error for version 2.3.0.
But if I remove all unnecessary `require`s, script works fine!

----------------------------------------------------------------------

We use browser Chrome 65.0.3325.181
And you can see our fork with [corrected](https://github.com/aleksandr-oleynikov/mocha-teamcity-reporter/blob/browser/lib/teamcity.js) `teamcity.js` to use it in browser.
Obviously it won't work in node.js, but you can see changes we made to fix this problem.

**Comment by aleksandr-oleynikov** [comment]
Actually we don't use flowId. So I think you can base it on `document.title` as a first solution.

**Comment by jamie-sherriff** [comment]
I have updated the run-in-browser and just split it out into a separate file to not need adding webpack for the moment, may add webpack in the future. i will do some more tidy up and merge this in soon.

**Comment by jamie-sherriff** [comment]
Fixed in 2.4.0
